Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp

```diff
@@ -1266,6 +1267,30 @@ namespace EMotionFX
         {
             m_floatData[rootJointDataIndex] = m_floatData[sampleJointDataIndex];
         }
+
+        if (data.m_smoothingMethod == RootMotionExtractionData::SmoothingMethod::MovingAverage)
+        {
+            SmoothData(true, false, data.m_smoothFrameNum);
+        }
+    }
+
+    void NonUniformMotionData::SmoothData(bool smoothPosition, bool smoothRotation, size_t smoothFrameNum)
```

Comment:
suggestion: Maybe using an enum bit mask here might be nicer than two bools? I'd look into the type safe enum thing too that I think @kberg-amzn added. Could be useful in this case to avoid using an unscoped enum

Reply:
good point, I changed it to use the rootMotionExtractionData directly, so it avoid this problem.